Sources/Public/BacktraceBreadcrumb.swift

```diff
@@ -0,0 +1,163 @@
+        case .navigation:
+            return "navigation"
+        case .http:
+            return "http"
+        case .system:
+            return "system"
+        case .user:
+            return "user"
+        case .configuration:
+            return "configuration"
+        }
+    }
+    
+    public static let all: [BacktraceBreadcrumbType] = [.manual, .log, .navigation, .http, .system, .user, .configuration]
+    
+    public static let none: [BacktraceBreadcrumbType] = []
+}
+
+@objc public enum BacktraceBreadcrumbLevel: Int {
+    
+    case debug = 1
+    case info = 2
+    case warning = 3
+    case http = 4
+    case error = 5
+    case fatal = 6
+    
+    var info : String {
+        switch self {
+        case .debug:
+            return "debug"
+        case .info:
+            return "info"
+        case .warning:
+            return "warning"
+        case .http:
+            return "http"
+        case .error:
+            return "error"
+        case .fatal:
+            return "fatal"
+        }
+    }
+    
+    public static let all: [BacktraceBreadcrumbLevel] = [.debug, .info, .warning, .http, .error, .fatal]
+    
+    public static let none: [BacktraceBreadcrumbLevel] = []
+}
+
+@objc public protocol BacktraceBreadcrumbProtocol {
+    @objc func addBreadcrumb(_ message: String,
+                        attributes: [String:Any],
+                        type: BacktraceBreadcrumbType,
+                        level: BacktraceBreadcrumbLevel) -> Bool
+    @objc func addBreadcrumb(_ message: String) -> Bool
+    @objc func addBreadcrumb(_ message: String,
+                        attributes: [String:Any]) -> Bool
+    @objc func addBreadcrumb(_ message: String,
+                        type: BacktraceBreadcrumbType,
+                        level: BacktraceBreadcrumbLevel) -> Bool
+    @objc func addBreadcrumb(_ message: String,
+                        level: BacktraceBreadcrumbLevel) -> Bool
+    @objc func addBreadcrumb(_ message: String,
+                        type: BacktraceBreadcrumbType) -> Bool
+}
+
+@objc public class BacktraceBreadcrumb: NSObject {
+    
+    private var enabledBreadcrumbTypes = [BacktraceBreadcrumbType]()
+        
+    private static let breadcrumbLogFileName = "bt-breadcrumbs-0"
+
+    public static var defaultMaxLogSize = 64000;
+#if os(iOS)
+    private var breadcrumbsLogManager: BacktraceBreadcrumbsLogManager?
+    private var backtraceComponentListener: BacktraceComponentListener?
+#endif
+    private func breadcrumbLogPath() throws -> String {
+        var fileURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+        fileURL.appendPathComponent(BacktraceBreadcrumb.breadcrumbLogFileName)
+        return fileURL.path
+    }
+    
+    public func enableBreadCrumbs(_ breadCrumbTypes: [BacktraceBreadcrumbType] = BacktraceBreadcrumbType.all , maxLogSize: Int = defaultMaxLogSize) {
+        do {
+#if os(iOS)
```

Comment:
on Mac/tvOS/other OS than iOS - does this method do anything?  If no - should we give a user a way to initialize breadcrumbs at all on os different than iOS?

Reply:
Currently Mac and tvOS is out of scope - so on these platforms it's supposed to do nothing for this beta version.